### ISSUE TITLE
Scope out additional loss functions

### ISSUE THREAD

**Issue by AnzeXie** [marius-team/marius#40]
**Is your feature request related to a problem? Please describe.**
Currently, the supported loss functions we have for Marius are SoftMax and RankingLoss. 

**Describe the solution you'd like**
We can expand the set of loss functions by implementing additional loss functions to Marius. 
We can implement losses into 2 new source files: loss.cpp and loss.h.
We can also add a new section in the configuration for loss options.

We can use the loss functions implemented by PyKeen as a reference:
List:
https://github.com/pykeen/pykeen#losses-7
Implementation:
https://github.com/pykeen/pykeen/blob/master/src/pykeen/losses.py
Documentation:
https://pykeen.readthedocs.io/en/stable/reference/losses.html

**Comment by JasonMoho** [comment]
We should add a new doc page and describe each loss function in it and also update the configuration documentation.
